clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java

```diff
@@ -332,6 +333,14 @@ public boolean isVersionSupported(short apiVersion) {
         return apiVersion >= oldestVersion() && apiVersion <= latestVersion();
     }
 
+    public short headerVersion(short apiVersion) {
+        if ((this == CONTROLLED_SHUTDOWN) && (apiVersion == 0)) {
```

Comment:
nit: I would rather overload the method to handle this case rather like we do for the parsing of `ApiVersions` here: https://github.com/apache/kafka/blob/trunk/clients/src/main/java/org/apache/kafka/common/protocol/ApiKeys.java#L154

Reply:
Why do we need this check? There seem to be a comment here that goes into some detail: https://github.com/apache/kafka/pull/7372/files#diff-532a8e555895b9ac3028f30a263b1a57L163-L165. Even after reading that comment it is not clear to me why we are doing this. Can we add a comment here that explains this?

Reply:
@jsancio : The bigger context here is that KIP-482 (and KIP-511) add a new header version, v2.  The new header version adds optional (tagged) fields.  This PR lays the groundwork for that, although this PR only has v0 and v1.

@dajac: In a follow-on PR, we will be calling into some automatically generated code to determine what the header version is.  This is part of the header version bump described in KIP-482.  Therefore, creating a subclass of `ApiKeys` is not a viable approach here.